src/Tools/SelectionTool.js

```diff
@@ -19,26 +19,32 @@ class SelectionTool extends BaseTool {
     super(appData)
 
     this.dragging = false
+    this.selectionManager = appData.selectionManager
 
     this.selectionRect = new Rect(1, 1)
     this.selectionRectMat = new Material('marker', 'ScreenSpaceShader')
     this.selectionRectMat.getParameter('BaseColor').setValue(new Color('#03E3AC'))
     this.selectionRectXfo = new Xfo()
     this.selectionRectXfo.tr.set(0.5, 0.5, 0)
     this.selectionRectXfo.sc.set(0, 0, 0)
+
+    this.rectItem = new GeomItem('selectionRect', this.selectionRect, this.selectionRectMat)
+    this.rectItem.getParameter('Visible').setValue(false)
+    this.appData.renderer.addTreeItem(this.rectItem)
+  }
+
+  /**
+   * Activates selection tool.
+   */
+  setSelectionManager(selectionManager) {
+    this.selectionManager = selectionManager
   }
```

Comment:
Note: appData is now optional. We can simple create the SelectionTool and then set the SelectionManager. I think this process is cleaner. We should migrate away from using appData everywhere like this.

Reply:
Indeed, I don't think we should use `appData` approach, we can either pass the `selectionManager` directly to the constructor or set it with the setter method. But line 22 isn't a robust implementation